src/Planners/src/SwingFootPlanner.cpp

```diff
@@ -157,23 +169,33 @@ bool SwingFootPlanner::setContactList(const ContactList& contactList)
     auto activeContactNewList = contactList.getActiveContact(m_currentTrajectoryTime);
     auto activeContact = m_contactList.getActiveContact(m_currentTrajectoryTime);
 
-    constexpr double tolerance = 1e-6;
     if (activeContact != m_contactList.cend() // i.e., the original list contains an active contact
         && activeContactNewList != contactList.cend() // i.e., the new list contains an active
                                                       // contact
     )
     {
-        if (!(activeContact->pose - activeContactNewList->pose).coeffs().isZero(tolerance))
+        if (!(activeContact->pose.translation() - activeContactNewList->pose.translation())
+                 .isZero(m_positionTolerance)
+            || !(activeContact->pose.asSO3() - activeContactNewList->pose.asSO3())
+                    .coeffs()
+                    .isZero(m_orientationTolerance))
```

Comment:
Is ``asSO3`` returning a rotation matrix? If so, the rotation tolerance is not really an angle, right?

Reply:
Rather than the norm, what about using 
```math
\arccos\left(\frac{\text{trace}\left(R_1^\top R_2\right) - 1}{2}\right)
```

See https://en.wikipedia.org/wiki/Rotation_matrix#Determining_the_angle

Reply:
I think its the same 

The equation you provide computes the angle of the angle axis representation, on the other hand, in case of SO(3) the log returns the axis angle representation ( see also
Section 2.1. in https://cvg.cit.tum.de/_media/members/demmeln/nurlanov2021so3log.pdf)

See: 
- https://en.wikipedia.org/wiki/Axis%E2%80%93angle_representation#Log_map_from_SO(3)_to_%F0%9D%94%B0%F0%9D%94%AC(3)
- Example 4 of https://arxiv.org/pdf/1812.01537.pdf

Reply:
Ah sorry, I did not understand that you wanted to compute the norm of the log and not the norm of the difference. That makes sense then

Reply:
ops :( sorry my fault!